addons/mail/models/mail_render_mixin.py

```diff
@@ -203,15 +206,20 @@ def _prepend_preview(self, html, preview):
         if preview:
             preview = preview.strip()
 
+        template_instructions = parse_inline_template(preview or '')
+        preview_markup = Markup('')
+        for string, expression in template_instructions:
```

Comment:
Ok, I understand it a bit better. It would probably gain from having a block-based rendering method allowing to hide some details. The whole injection of expression should be tested, and probably done in a specific method in tools to ensure we it can be tested and validated independently from the prepend tool method itself.